models/MITgcm_ocean/trans_mitdart.f90

```diff
@@ -0,0 +1,852 @@
+   call check(nf90_put_att(ncid, dic_varid, "units_long_name", "moles Carbon per cubic meters"))
+   
+   ! 7. BLING tracer: dissolved organic phosphorus DOP
+   call check(nf90_def_var(ncid=ncid, name="DOP", xtype=nf90_real, &
+        dimids=(/XCDimID,YCDimID,ZCDimID/),varid=dop_varid))
+   call check(nf90_put_att(ncid, dop_varid, "long_name"      , "Dissolved Organic Phosphorus"))
+   call check(nf90_put_att(ncid, dop_varid, "missing_value"  , FVAL))
+   call check(nf90_put_att(ncid, dop_varid, "_FillValue"     , FVAL))
+   call check(nf90_put_att(ncid, dop_varid, "units"          , "mol P/m3"))
+   call check(nf90_put_att(ncid, dop_varid, "units_long_name", "moles Phosphorus per cubic meters"))
+   
+   ! 8. BLING tracer: dissolved organic nitrogen DON
+   call check(nf90_def_var(ncid=ncid, name="DON", xtype=nf90_real, &
+        dimids=(/XCDimID,YCDimID,ZCDimID/),varid=don_varid))
+   call check(nf90_put_att(ncid, don_varid, "long_name"      , "Dissolved Organic Nitrogen"))
+   call check(nf90_put_att(ncid, don_varid, "missing_value"  , FVAL))
+   call check(nf90_put_att(ncid, don_varid, "_FillValue"     , FVAL))
+   call check(nf90_put_att(ncid, don_varid, "units"          , "mol N/m3"))
+   call check(nf90_put_att(ncid, don_varid, "units_long_name", "moles Nitrogen per cubic meters"))
+   
+   ! 9. BLING tracer: dissolved inorganic iron FET
+   call check(nf90_def_var(ncid=ncid, name="FET", xtype=nf90_real, &
+        dimids=(/XCDimID,YCDimID,ZCDimID/),varid=fet_varid))
+   call check(nf90_put_att(ncid, fet_varid, "long_name"      , "Dissolved Inorganic Iron"))
+   call check(nf90_put_att(ncid, fet_varid, "missing_value"  , FVAL))
+   call check(nf90_put_att(ncid, fet_varid, "_FillValue"     , FVAL))
+   call check(nf90_put_att(ncid, fet_varid, "units"          , "mol Fe/m3"))
+   call check(nf90_put_att(ncid, fet_varid, "units_long_name", "moles Iron per cubic meters"))
+   
+   ! 10. BLING tracer: Surface Chlorophyl CHL
+   call check(nf90_def_var(ncid=ncid, name="CHL", xtype=nf90_real, &
+        dimids=(/XCDimID,YCDimID/),varid=chl_varid))
+   call check(nf90_put_att(ncid, chl_varid, "long_name"      , "Surface Chlorophyll"))
+   call check(nf90_put_att(ncid, chl_varid, "missing_value"  , FVAL))
+   call check(nf90_put_att(ncid, chl_varid, "_FillValue"     , FVAL))
+   call check(nf90_put_att(ncid, chl_varid, "units"          , "mg/m3"))
+   call check(nf90_put_att(ncid, chl_varid, "units_long_name", "milligram per cubic meters"))
+endif   
+
+! Finished with dimension/variable definitions, must end 'define' mode to fill.
+
+call check(nf90_enddef(ncid))
+
+! Fill the coordinate variables
+
+call check(nf90_put_var(ncid, XGVarID, XG ))
+call check(nf90_put_var(ncid, XCVarID, XC ))
+call check(nf90_put_var(ncid, YGVarID, YG ))
+call check(nf90_put_var(ncid, YCVarID, YC ))
+call check(nf90_put_var(ncid, ZCVarID, ZC ))
+
+! Fill the data
+
+iunit = get_unit()
+open(iunit, file='PSAL.data', form='UNFORMATTED', status='OLD', &
```

Comment:
i see this program came from someone else.  however, in other parts of the code, recl is set by a line like:

```
reclen = product(shape(x)) * item_size_direct_access
```

for write (since you know the size of array x).  and for read,

```
reclen = metadata%reclen * item_size_direct_access
```

where i'm not sure where reclen comes from but it looks like a namelist or is read from some metadata source. 

at a minimum the magic number 4 needs to be replaced with a parameter, and
```
product(shape(data_3d))
```
might be more clear than Nx, Ny, Nz.  (you can argue with me on that.)